tools/CheckH5.py

```diff
@@ -0,0 +1,207 @@
+    # which `abs` returns a float
+    def approx(self, test_data, expected_data):
+        return abs(test_data - expected_data) < self.absolute_tolerance and (
+            max(abs(test_data), abs(expected_data)) == 0.0 or
+            self.relative_tolerance is None or abs(test_data - expected_data) /
+            max(abs(test_data), abs(expected_data)) < self.relative_tolerance)
+
+    # Perform the unit test comparisons between the dataset or group
+    # represented by the string `test_entity` and `expected_entity`.
+    # If `expected_entity` is `None`, then `test_entity` (or, if it is a group,
+    # the datasets within the group) is compared to 0.0.
+    def check_h5_file(self, h5_file, test_entity, expected_entity):
+        if isinstance(h5_file[test_entity], h5py.Dataset):
+            # It is recommended to include subTest commands once python3 can be
+            # required:
+            # with self.unit_test.subTest(test_entity=test_entity,
+            #                             expected_entity=expected_entity):
+            test_data = h5_file[test_entity][()]
+            if self.expected_h5_entity is not None:
+                expected_data = h5_file[expected_entity][()]
+                self.unit_test.assertEqual(
+                    test_data.shape, expected_data.shape,
+                    "test and expected h5 datasets must have identical"
+                    " sizes.")
+                self.unit_test.assertEqual(
+                    test_data.dtype, expected_data.dtype,
+                    "test and expected h5 datasets must have identical"
+                    " types.")
+                if (test_data.dtype == np.dtype(np.float)
+                        or test_data.dtype == np.dtype(np.complexfloating)):
+                    for index in np.ndindex(test_data.shape):
+                        # It is recommended to include subTest commands once
+                        # python3 can be required:
+                        # with self.unit_test.subTest(index=index):
+                        self.unit_test.assertTrue(
+                            self.approx(test_data[index],
+                                        expected_data[index]),
+                            msg="Compare " + str(test_data[index]) + " to " +
+                            str(expected_data[index]))
+                else:
+                    self.unit_test.assertEqual(test_data, expected_data)
+            else:
+                if (test_data.dtype == np.dtype(np.floating)
+                        or test_data.dtype == np.dtype(np.complexfloating)):
+                    for index in np.ndindex(test_data.shape):
+                        # It is recommended to include subTest commands once
+                        # python3 can be required:
+                        # with self.unit_test.subTest(index=index):
+                        self.unit_test.assertTrue(
+                            self.approx(test_data[index], 0.0))
```

Comment:
since we have an absolute and relative tolerance why not compare using that? Maybe I don't understand the use-case here

Reply:
Sorry, I'm confused by this comment -- the approx method uses the absolute and relative tolerance, this is just here to avoid code duplication.
A test that gives only a single value to compare against zero doesn't give a good measure of scale, so currently will use only the absolute tolerance value and the user is expected to set that to something reasonable for their problem. I can change that behavior, though, if you think something else would be better suited.

Reply:
yea, just ignore it, I didn't read well :(